base/array.jl

```diff
@@ -2436,6 +2436,48 @@ function filter!(f, a::AbstractVector)
     return a
 end
 
+"""
+     mask!(a::AbstractVector, m::AbstractVector{Bool})
+
+The inplace version of logical indexing `a = a[m]`. That is, `mask!(a, m)` on
+vectors of equal length `a` and `m` will remove all elements from `a` for which
+`m` at the corresponding index is `false`.
+
+# Examples
+```jldoctest
+julia> a = [:a, :b, :c];
+
+julia> mask!(a, [true, false, true])
+2-element Vector{Symbol}:
+ :a
+ :c
+
+julia> a
+2-element Vector{Symbol}:
+ :a
+ :c
+```
+"""
+function mask!(a::AbstractVector, m::AbstractVector{Bool})
+    j = firstindex(a)
+    for i in eachindex(a, m)
+        @inbounds begin
+            ai = a[i]
+            mi = m[i]
+            a[j] = ai
+        end
+        j = ifelse(mi, nextind(a, j), j)
+    end
+    j > lastindex(a) && return a
+    if a isa Vector
+        resize!(a, j-1)
+        sizehint!(a, j-1)
+    else
+        deleteat!(a, j:lastindex(a))
+    end
```

Comment:
resize! and deleteat! call the same underlying method, so there's not really any reason to split the code path
```suggestion
    deleteat!(a, j:lastindex(a))
```

Reply:
This is also something I lifted from the implementation of `filter!`. Should this be changed in `filter!` as well?